subprojects/file-watching/src/main/java/org/gradle/internal/watch/registry/impl/HierarchicalFileWatcherUpdater.java

```diff
@@ -82,35 +91,61 @@ protected boolean handleVirtualFileSystemContentsChanged(Collection<FileSystemLo
     }
 
     @Override
-    protected SnapshotHierarchy doUpdateVfsOnBuildStarted(SnapshotHierarchy root) {
-        return movedHierarchyHandler.handleMovedHierarchies(root);
+    public SnapshotHierarchy updateVfsOnBuildFinished(SnapshotHierarchy root, WatchMode watchMode, int maximumNumberOfWatchedHierarchies) {
+        SnapshotHierarchy newRoot = super.updateVfsOnBuildFinished(root, watchMode, maximumNumberOfWatchedHierarchies);
+        LOGGER.info("Watched directory hierarchies: {}", watchedHierarchies);
```

Comment:
I think it would make sense to log the `getNumberOfWatchedLocations()` at the end of the build anyway, so we could move this logging to the base class and then have one less overridden method. WDYT?

Reply:
Good point. We are already logging this at the end of the build here with verbose logging:

https://github.com/gradle/gradle/blob/afcc0fd0a34bd8279d9b63fb727f40280fbdbdce/subprojects/file-watching/src/main/java/org/gradle/internal/watch/vfs/impl/WatchingVirtualFileSystem.java#L203-L205

I think that should be enough. WDYT? I'll remove this one here.

Reply:
Actually, that messes with some of the tests. I'd rather do this in a separate PR that handles logging.